hist/hist/inc/TF1.h

```diff
@@ -783,4 +790,103 @@ void TF1::SetFunction(PtrObj &p, MemFn memFn)
    fFunctor = new TF1::TF1FunctorPointerImpl<double>(ROOT::Math::ParamFunctor(p, memFn));
 }
 
+template <class T>
+inline T TF1::GradientPar(Int_t ipar, const T *x, Double_t eps)
+{
+   if (fType == EFType::kTemplated) {
+      return GradientParVec<T>(ipar, x, eps);
+   } else
+      return TF1::GradientPar(ipar, (const Double_t *)x, eps);
+}
+
+template <class T>
+inline T TF1::GradientParVec(Int_t ipar, const T *x, Double_t eps)
+{
+   if (GetNpar() == 0) return 0;
+
+   if (eps < 1e-10 || eps > 1) {
+      Warning("Derivative", "parameter esp=%g out of allowed range[1e-10,1], reset to 0.01", eps);
+      eps = 0.01;
+   }
+   Double_t h;
+   TF1 *func = (TF1 *)this;
+   Double_t *parameters = GetParameters();
+
+   // If we are in a multi-threading scenario, make this function thread-safe
+   // using a copy of the parameters vector.
+   // TODO: Check that GetImplicitMTPoolSize is the best method to check that we
+   // have more than one thread accesing the resources.
+   if (ROOT::GetImplicitMTPoolSize() != 0) {
+      std::vector<Double_t> parametersCopy(GetNpar());
+      std::copy(parameters, parameters + GetNpar(), parametersCopy.begin());
+      parameters = parametersCopy.data();
+   }
+
+   // func->InitArgs(x, parameters);
+   if (fMethodCall) {
+      Error("TF1::GradientParVec<T>", "Vectorized implementation does not support interpreted functions");
+   }
+
+   Double_t al, bl;
+   T f1, f2, g1, g2, h2, d0, d2;
+
+   ((TF1 *)this)->GetParLimits(ipar, al, bl);
+   if (al * bl != 0 && al >= bl) {
+      // this parameter is fixed
+      return 0;
+   }
+
+   // check if error has been computer (is not zero)
+   if (func->GetParError(ipar) != 0)
+      h = eps * func->GetParError(ipar);
+   else
+      h = eps;
+
+   // save original parameters
+   Double_t par0 = parameters[ipar];
+
+   parameters[ipar] = par0 + h;
+   f1 = func->EvalPar(x, parameters);
+   parameters[ipar] = par0 - h;
+   f2 = func->EvalPar(x, parameters);
+   parameters[ipar] = par0 + h / 2;
+   g1 = func->EvalPar(x, parameters);
+   parameters[ipar] = par0 - h / 2;
+   g2 = func->EvalPar(x, parameters);
+
+   // compute the central differences
+   h2 = 1 / (2. * h);
```

Comment:
Why is h2 not a Double_t? if h is a Double_t (as it is right now) then h2 should keep the type, at least for the use you're making of it.